src/notes/FluxNotesEditor.jsx

```diff
@@ -302,13 +308,21 @@ class FluxNotesEditor extends React.Component {
         if (!this.shortcutTriggerCheck(shortcutC, shortcutTrigger)) {
             return this.insertPlainText(transform, shortcutTrigger);
         }
-
         let shortcut = this.props.newCurrentShortcut(shortcutC, shortcutTrigger, text, updatePatient, source);
+        if(text === "" && shortcut.needToSelectValueFromMultipleOptions()){
+            shortcut.setText(shortcutTrigger);
+        }
         shortcut.initialContextPosition = initialContextPosition;
+        
+        this.insertStructuredFieldTransform(transform, shortcut).collapseToStartOfNextText().apply();
```

Comment:
[DP2] We shouldn't be applying our transform if we are going to continue modifying that transform later on, as we conditionally do on line 323.